test/login_test.dart

```diff
@@ -7,7 +9,10 @@ import 'config.dart';
 void main() {
   group('Login', () {
     test('Login flow works', () async {
-      var client = NextCloudClient.withoutLogin(Config.host);
+      var client = NextCloudClient.withoutLogin(
+        Config.host,
+        defaultHeaders: {HttpHeaders.userAgentHeader: 'dart-nextcloud'},
```

Comment:
I haven't really added any new tests that specifically test the header behavior, yet. Just wanted to get your opinion before I start pulling in stuff like mockito.